test/extended/builds/failure_status.go

```diff
@@ -174,14 +174,32 @@ var _ = g.Describe("[sig-builds][Feature:Builds][Slow] update failure status", f
 				var build *buildv1.Build
 				wait.PollImmediate(200*time.Millisecond, 30*time.Second, func() (bool, error) {
 					build, err = oc.BuildClient().BuildV1().Builds(oc.Namespace()).Get(context.Background(), br.Build.Name, metav1.GetOptions{})
-					if build.Status.Reason != buildv1.StatusReasonOutOfMemoryKilled {
+					// In 4.15, status reason may be filed as Error rather than OOMKilled
+					// (i.e. https://github.com/containerd/containerd/issues/8180) and to unblock the test
+					// we can also rely on the exit code 137 which corresponds to OOMKilled.
+					if build.Status.Reason != buildv1.StatusReasonOutOfMemoryKilled && build.Status.Reason != buildv1.StatusReasonGenericBuildFailed {
 						return false, nil
 					}
 					return true, nil
 				})
 				o.Expect(err).NotTo(o.HaveOccurred())
-				o.Expect(build.Status.Reason).To(o.Equal(buildv1.StatusReasonOutOfMemoryKilled))
-				o.Expect(build.Status.Message).To(o.Equal("The build pod was killed due to an out of memory condition."))
+				o.Expect(build.Status.Reason).To(o.Or(o.Equal(buildv1.StatusReasonOutOfMemoryKilled), o.Equal(buildv1.StatusReasonGenericBuildFailed)))
+				if build.Status.Reason == buildv1.StatusReasonOutOfMemoryKilled {
+					o.Expect(build.Status.Message).To(o.Equal("The build pod was killed due to an out of memory condition."))
+				} else {
+					pod, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).Get(context.Background(), build.Name+"-build", metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+					oomKilledExitCodeFound := false
+					for _, c := range pod.Status.ContainerStatuses {
+						if c.State.Terminated == nil {
+							continue
+						}
+						if c.State.Terminated.ExitCode == 137 {
+							oomKilledExitCodeFound = true
+						}
+					}
+					o.Expect(oomKilledExitCodeFound).To(o.BeTrue())
+				}
```

Comment:
Won't we expect exit code 137, regardless of status? If so, then I don't think we need the if/else.